src/ufuncs.c

```diff
@@ -1535,6 +1535,65 @@ static void relate_func(char** args, npy_intp* dimensions, npy_intp* steps, void
 }
 static PyUFuncGenericFunction relate_funcs[1] = {&relate_func};
 
+static char relate_pattern_dtypes[4] = {NPY_OBJECT, NPY_OBJECT, NPY_OBJECT, NPY_BOOL};
+static void relate_pattern_func(char** args, npy_intp* dimensions, npy_intp* steps,
+                                void* data) {
+  GEOSGeometry *in1 = NULL, *in2 = NULL;
+  const char* pattern;
+  npy_bool ret;
+
+  /* get the pattern argument (only deal with scalar for now) */
+  char* ip3 = args[2];
+  npy_intp is3 = steps[2];
+
+  if (is3 != 0) {
+    PyErr_Format(PyExc_ValueError, "pattern keyword only supports scalar argument");
+    return;
+  }
+  PyObject* in3 = *(PyObject**)ip3;
+  if (PyUnicode_Check(in3)) {
+    pattern = PyUnicode_AsUTF8(in3);
+    if (pattern == NULL) {
+      return;
```

Comment:
This seems like it indicates an error condition, but isn't reported back as such?  Should this raise an exception?

Reply:
My understanding is that the error is set by `PyUnicode_AsUTF8`. 
Now, I don't know what case could trigger this (or if it even possible from python), so not sure how to test it.

Reply:
Right
> In the case of an error, NULL is returned with an exception set

Perhaps a comment would be helpful to note that it set the exception and that `NULL` here indicates a bad string?

Reply:
Added a comment